File: CsChatClient/Messages/TextMessageContent.cs
namespace CsChatClient.Messages
{
    [ContentAttribute(MessageContentType.MessageContentTypeText, MessageContentPersistFlag.PersistFlagPersistAndCount)]
    public class TextMessageContent : MessageContent
    {
        public string Content { get; set; }

        public override void Decode(MessagePayload payload)
        {
            Content = payload.SearchableContent;    
        }

        public override string Digest(MessageEx message)
        {
            return Content;
        }

        public override MessagePayload Encode()
        {
            MessagePayload payload = new MessagePayload();
            payload.SearchableContent = Content;
            return payload;
        }
    }
}

File: CppChatClient/message/TextMessageContent.cpp
#include "TextMessageContent.h"

namespace WFCLib {

TextMessageContentPrototype TextMessageContent::sPrototype;

MessagePayload TextMessageContent::encode() const
{
	MessagePayload payload = MessageContent::encode();
    
	payload.searchableContent = content;
    payload.mentionedType = mentionedType;
    payload.mentionedTargets = mentionedTargets;
    
	return payload;
}

void TextMessageContent::decode(const MessagePayload & payload)
{
    MessageContent::decode(payload);
    content = payload.searchableContent;
    mentionedType = payload.mentionedType;
    mentionedTargets = payload.mentionedTargets;
}

MessageContent* TextMessageContent::clone() const {
    TextMessageContent *p = new TextMessageContent();
    p->extra = extra;
    p->content = content;
    p->mentionedType = mentionedType;
    p->mentionedTargets = mentionedTargets;
    return p;
}

const MessageContentPrototype* TextMessageContent::getPrototype() const {
    return &sPrototype;
}

std::string TextMessageContent::digest() const {
    return content;
}

};
